Updating Fusion thread information (#8554)
fishtown-analytics/docs.getdbt.com: website/snippets/_fusion-threads.md

> | Adapter | Behavior |

Reviewer: this table is helpful! is there a way we can create another column to this table so it's less redundant and separates the 'notes' section a bit? for example:

adapter | who controls threads? | notes
|-- | -- | --|
snowflake | fusion-controlled | user threads ignored; fusion optimizes parallelism automatically. only threads: 1 respected.
databricks | fusion-controlled | user threads ignored; fusion optimizes parallelism automatically. only threads: 1 respected.
bq | user-controlled | user threads respected to manage API rate limits. threads: 0 enables dynamic optimization.
redshift | user-controlled | user threads respected to manage warehouse concurrency. threads: 0 enables dynamic optimization.

Author: I don't think that's entirely accurate as the user _can_ control the threads on BQ and redshift but Fusion can as well, it doesn't have to be user defined

Reviewer: ok got it. i think we say the same thing in the current content (fusion respects user set..) but we're asking the user to then read below and piece together that fusion dynamically takes over _if_ threads are set to 0. we can also change the 2nd column from user-controlled to user or fusion controlled or maybe the header shoudl be 'thread default' or something. 

not a huge blocker but i personally think it's asking the user to stitch things together and reconcile multiple sentences when we can make it easier for them to parse.

|---------|----------|
| **Snowflake** | Fusion ignores user-set threads and automatically optimizes parallelism for maximum performance. |
| **Databricks** | Fusion ignores user-set threads and automatically optimizes parallelism for maximum performance. |
| **BigQuery** | Fusion respects user-set threads to manage API rate limits. |
| **Redshift** | Fusion respects user-set threads to manage concurrency limits. |

> For BigQuery and Redshift, setting `--threads 0` (or omitting the setting) allows Fusion to dynamically optimize parallelism.

Reviewer: sugg bullets and starting with snowflake/dbx to match table
```suggestion
- For Snowflake and Databricks, <Constant name="fusion"/>  manages parallelism automatically and ignores user-set thread values. The only supported override is `threads: 1`, which can also help resolve timeout issues if set. 
```

> For Snowflake and Databricks, only `threads: 1` is respected. If you are seeing timeouts due to Fusion parallelism, try setting `threads: 1` to see if this resolves the issue.

Reviewer: ```suggestion
- For BigQuery and Redshift, <Constant name="fusion"/> respects user-set thread values to accommodate warehouse-specific limits. Setting `--threads 0` (or omitting the setting) allows <Constant name="fusion"/> to dynamically optimize parallelism.
```

| Adapter | Behavior |
|---------|----------|
| **Snowflake** | Fusion ignores user-set threads and automatically optimizes parallelism for maximum performance. |
| **Databricks** | Fusion ignores user-set threads and automatically optimizes parallelism for maximum performance. |
| **BigQuery** | Fusion respects user-set threads to manage API rate limits. |
| **Redshift** | Fusion respects user-set threads to manage concurrency limits. |

> For Snowflake and Databricks, setting `--threads 0` (or omitting the setting) allows Fusion to dynamically optimize parallelism. Low thread values (like 4 or 6) can significantly slow down Fusion performance on these platforms.

Author: ```suggestion
For BigQuery and Redshift, setting `--threads 0` (or omitting the setting) allows Fusion to dynamically optimize parallelism.

For Snowflake and Databricks, only `threads: 1` is respected. If you are seeing timeouts due to Fusion parallelism, try setting `threads: 1` to see if this resolves the issue.
```
